components/esp_http_client/esp_http_client.c

```diff
@@ -632,6 +632,12 @@ esp_http_client_handle_t esp_http_client_init(const esp_http_client_config_t *co
         }
     }
 
+    if (config->client_key_password) {
+        if(config->client_key_password_len > 0){
```

Comment:
`config->client_key_password & config->client_key_password_len > 0`

Could be a single if statement.